src/condor_dagman/dagman_submit.cpp

```diff
@@ -112,6 +112,7 @@ static std::vector<NodeVar> init_vars(const Dagman& dm, const Job& node) {
 		batchId = dm.options[deep::str::BatchId];
 	}
 
+	vars.push_back(NodeVar("NODE_NAME", nodeName, false));
```

Comment:
FWIW, this is one place where it makes a (small) difference to use `emplace_back` rather than `push_back`.  `emplace_back` uses magic perfect forwarding, so you can do

```
vars.emplace_back("NODE_NAME", nodeName, false);
```

rather than create a NodeVar, and copy it into the vector